Sources/TensorFlow/Layer.swift

```diff
@@ -191,3 +191,35 @@ public final class Parameter<Scalar: TensorFlowScalar> {
         self.value = value
     }
 }
+
+/// A wrapper around a differentiable value with toggleable trainability.
+@_propertyWrapper
+public struct Trainable<Value: Differentiable> : Differentiable {
+    @noDerivative public var isTrainable: Bool = true
+    public var _value: Value
```

Comment:
Why not make `value` a stored property and get rid of `_value`?

Reply:
I don't believe `@differentiable` on stored properties supports custom derivatives:
```
/Users/danielzheng/swift-apis/Sources/TensorFlow/Layer.swift:199:6: error: 'jvp:' or 'vjp:' cannot be specified for stored properties
    @differentiable(vjp: _vjpValue)
    ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```